remote-client/src/main/java/org/opensearch/remote/metadata/client/impl/RemoteClusterIndicesClient.java

```diff
@@ -564,7 +562,7 @@ private XContentParser createParser(JsonpSerializable obj) throws IOException {
         try (JsonGenerator generator = mapper.jsonProvider().createGenerator(stringWriter)) {
             mapper.serialize(obj, generator);
         }
-        return jsonXContent.createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.IGNORE_DEPRECATIONS, stringWriter.toString());
+        return SdkClientUtils.createParser(stringWriter.toString());
```

Comment:
I assume DDB client parsing logic is different which is why we aren't adding the same changes to the ddb client?

Reply:
Actually DDB client logic is identical and we should make this same change there.

This isn't so much a change as a refactoring, moving the `jsonXContent.createParser` code to the `SdkClientUtils` class and delegating the call there.

The reason it was less important for the bug fix is that DDB doesn't implement search.